### PR TITLE
Refactor facade

### PR DESCRIPTION
**Pull request by GustavoZHA**

### CODE REVIEW

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java [hunk 1]
```diff
@@ -35,8 +34,8 @@
 @RequestMapping("/api")
 public class AudioConverterController {
     private static final Logger LOGGER = LogManager.getLogger();
-    private ParameterOutputChecksum paramChecksum;
-    private FileStorageService fileStorageService = new FileStorageService();
+    private ParameterOutputChecksum parameterOutputChecksum;
+    private final FileStorageService fileStorageService = new FileStorageService();
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java]: final variables must be uppercase

File: src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    private static String baseUrl = ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString();
+
+    private DownloadLinkFacade() {
+    }
+
+    /**
+     *  Generator link to download to convert.
+     *
+     * @param outputFilename is path of file which are converted.
+     * @return path which permit download the result process.
```
Comment (JorgeRCV) [inline src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java]: Maybe you can add the type of variable that returns, e.g. @return a String path which.......

File: src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java [hunk 3]
```diff
@@ -64,17 +64,16 @@ public String uploadFile(final @RequestParam("file") MultipartFile file,
...
-        String path = paramChecksum.getOutputFilename();
-        videoParam = new VideoParameter(path, outputFormat, resolution, thumbnail, frameRate, width, height, audio);
-        String outputFilename = ConverterFacade.getVideoConverter(videoParam);
+                             final @RequestParam("metadata") boolean metadata)
+            throws IOException, InterruptedException {
+        LOGGER.info("start");
+        parameterOutputChecksum = ChecksumFacade.getChecksum(checksum, file);
+        String outputFilename = ConverterFacade.getVideoConverter(
+                new VideoParameter(parameterOutputChecksum.getOutputFilename(), outputFormat, resolution, thumbnail,
+                        frameRate, width, height, audio));
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java:71]: These lines of code must be aligned and reach up to 120 characters

File: src/main/java/org/fundacion/jala/converter/core/facade/CompilerFacade.java [hunk 4]
```diff
@@ -26,13 +31,43 @@
...
+        }
+        return "";
+    }
+
+    /**
+     * Compiles a Python file.
+     *
+     * @param code is string with code in Python.
+     * @return a string of the result on runtime console.
+     */
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/core/facade/CompilerFacade.java]: in the ereturn javadoca a string must be uppercase to String

File: src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        String outputFileName = fileOut.substring(0, fileOut.lastIndexOf("."));
+        return baseUrl + "/api/download/" + outputFileName + ".txt";
+    }
+
+    /**
+     *
+     * @param filename is name of file which are extracted metadata.
+     * @param format is format which the result process do.
+     * @return path which permit download the result process.
+     */
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java]: the return of the javadoce must start with a String

example: @return a String path which permit download the result process

File: src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package org.fundacion.jala.converter.core.facade;
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java:12]: class copyright and author missing

File: src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+import org.fundacion.jala.converter.core.ExtractMetadata;
+import org.fundacion.jala.converter.core.FileStorageService;
+import java.io.File;
+
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java:17]: class javadoc is missing

File: src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java (hunk 3 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java:71]: I did, this configuration is given by intellij

File: src/main/java/org/fundacion/jala/converter/core/facade/CompilerFacade.java (hunk 4 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/core/facade/CompilerFacade.java]: you are rigth

File: src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java (hunk 2 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java]: I forgot it, thanks

File: src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java (hunk 5 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/core/facade/DownloadLinkFacade.java]: ok juampi

File: src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java (hunk 7 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/core/facade/MetadataFacade.java:17]: Ho no

**Review by jpgonzales27 — APPROVED** [review]

**Review by JorgeRCV — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/core/facade/ZipFileFacade.java [hunk 8]
```diff
@@ -28,65 +29,86 @@ private ZipFileFacade() {
...
-                                       final String storagePath) throws IOException, InterruptedException {
-        getZipFile(parameterOutputChecksum, metadata, storagePath, false);
+    public static void getZipFileAudio(final ParameterOutputChecksum parameterOutputChecksum, final boolean metadata,
+                                       final String outputFilename) throws IOException, InterruptedException {
+        builtZipFile(parameterOutputChecksum, metadata, outputFilename, false);
     }
 
     /**
-     * Compresses files of video.
+     * Compresses files of Video.
```
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/core/facade/ZipFileFacade.java:42]: Compresses files of "a" Video.

File: src/main/java/org/fundacion/jala/converter/core/facade/ParameterOutputChecksum.java [hunk 9]
```diff
@@ -39,7 +39,7 @@ public String getChecksumLocal() {
     /**
      * Gets Out put Filename.
      *
-     * @return string with out put Filename.
+     * @return a String with out put Filename.
```
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/core/facade/ParameterOutputChecksum.java:42]: Could be better described as: a String without the Filename output.

File: src/main/java/org/fundacion/jala/converter/controller/ExtractTextController.java [hunk 10]
```diff
@@ -38,23 +36,16 @@
      *
      * @param file is image file to extract text.
      * @param language is a type of language of the text.
-     * @return path to download files.
+     * @return a string of path to download files.
```
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/controller/ExtractTextController.java:39]: Should be: "a String of the path to download a file."

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java [hunk 11]
```diff
@@ -36,20 +36,18 @@
      * @param isMoreInfo is more information about the file.
      * @param nameExport is a name of file to export.
      * @param format is the format to file.
-     * @return path to download files.
+     * @return a string of path to download files.
```
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java:39]: Should be "a String of the path to download a metadata file."

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java [hunk 12]
```diff
@@ -49,32 +48,25 @@
      * @param audioChannel is the audioChannel with are converted of audio.
      * @param checksum is checksum of audio.
      * @param metadata if metadata is extracted from the audio.
-     * @return path to download files.
+     * @return a string of path to download files.
```
Comment (SaulCM) [inline src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java:51]: Should be: "a String of the path to download an audio file."

**Comment by JorgeRCV** [comment]
Great Job Gustavo,

Everthing it's ok but you could improve de redaction on javadocs in the next PRs

**Review by SaulCM — APPROVED** [review]

**Pull request merged**